hexrd/xrdutil/utils.py

```diff
@@ -1138,32 +1138,23 @@ def _project_on_detector_plane(allAngs,
                                rMat_d, rMat_c, chi,
                                tVec_d, tVec_c, tVec_s,
                                distortion,
-                               beamVec=constants.beam_vec):
+                               beam_vec=constants.beam_vec):
     """
     utility routine for projecting a list of (tth, eta, ome) onto the
     detector plane parameterized by the args
     """
-    gVec_cs = xfcapi.angles_to_gvec(allAngs,
-                                  chi=chi,
-                                  rmat_c=rMat_c,
-                                  beam_vec=beamVec)
-
-    rMat_ss = xfcapi.make_sample_rmat(chi, allAngs[:, 2])
+    # (angs, beam_vec, eta_vec, chi, rmat_c)
 
-    tmp_xys = xfcapi.gvec_to_xy(
-        gVec_cs, rMat_d, rMat_ss, rMat_c,
-        tVec_d, tVec_s, tVec_c,
-        beam_vec=beamVec)
+    tmp_xys = xfcapi.angles_to_gvec_to_xy_from_angles(chi, allAngs,
+                                                      rMat_d, rMat_c,
+                                                      tVec_d, tVec_s, tVec_c,
+                                                      beam_vec)
 
     valid_mask = ~(np.isnan(tmp_xys[:, 0]) | np.isnan(tmp_xys[:, 1]))
-
-    det_xy = np.atleast_2d(tmp_xys[valid_mask, :])
-
-    # apply distortion if specified
     if distortion is not None:
-        det_xy = distortion.apply_inverse(det_xy)
+        tmp_xys = distortion.apply_inverse(tmp_xys)
 
-    return det_xy, rMat_ss, valid_mask
+    return tmp_xys, None, valid_mask
```

Comment:
The second return is now `None`? We should verify it isn't being used anywhere, and we should look into potentially removing it as well.

Reply:
I see now that there is actually one place in hexrd which uses the second argument `rMat_ss`, [here](https://github.com/HEXRD/hexrd/blob/performance-tuning/hexrd/xrdutil/utils.py#L1549C17-L1549C23). I missed this in my assessment of this function earlier. I'll add this return value back in, excellent catch.

Reply:
@ZackAttack614 and I discussed this, and since `rMat_ss` is not normally used, we will no longer compute it in this function, but we will add a new function where it gets computed.